python/cudf/cudf/core/index.py

```diff
@@ -2960,6 +2940,7 @@ def as_index(arbitrary, **kwargs) -> Index:
 
 
 _dtype_to_index = {
+    bool: Index,
```

Comment:
Should this be `np.bool_` instead of `bool`?

Reply:
This was an error on my part and should be removed entirely. Before I figured out how to handle the different dtype->Index type conversions in `Index._copy_construct` correctly, I think adding this happened to fix a couple of test failures by causing a `KeyError` that was defaulting to the correct behavior.